### PR TITLE
Structure (Protobuf) - Aware Fuzzing Support for Apollo Unit Test

### PR DESCRIPTION
**Pull request by jiayunhan**
Add Protobuf-aware fuzzing support. Made the following changes:
- Add a external git repository [libprotobuf-mutator for Apollo](https://github.com/jiayunhan/libprotobuf-mutator) into WORKSPACE.in, which is a stable version that provides the mutation support for Protobuf messages. 
- Add lat_controller_fuzzer.cc, lon_controller_fuzzer.cc, simple_control_fuzz.cc, and some other fuzzing testcases as examples, together with their BUILD files. 
- Add a CROSSTOOL for clang-6.0, which is required to use libFuzzer. 
- Modified apollo.sh to exclude fuzzer unit test from the build command, since they rely on CROSSTOOL. The fuzzer unit test can only be built using bazel command manually. 

PASS all CPP Lint check.

Detailed introduction, and instructions for this pull request, can be found in this [README](https://github.com/jiayunhan/libprotobuf-mutator). 
If you have any questions, ping me on Baidu Hi. Thanks!

### CODE REVIEW

File: WORKSPACE.in [hunk 1]
```diff
@@ -203,3 +203,10 @@ new_local_repository(
     build_file = "third_party/adv_plat.BUILD",
     path = "/usr/local/apollo/adv_plat"
 )
+#libprotobuf-mutator
```
Comment (xiaoxq) [inline WORKSPACE.in]: Blank line before the section and new line after.

File: WORKSPACE.in [hunk 2]
```diff
@@ -203,3 +203,10 @@ new_local_repository(
...
+new_git_repository(
+    name = "libprotobuf_mutator",
+    build_file = "third_party/libprotobuf_mutator.BUILD",
+    commit = "91a872998bc204e9484dccdbeeafc608c1ee8062",
+    remote = "https://github.com/jiayunhan/libprotobuf-mutator",
```
Comment (xiaoxq) [inline WORKSPACE.in]: Better to refer an official repo instead of personal.

File: modules/control/control.h [hunk 3]
```diff
@@ -98,7 +98,7 @@ class Control : public apollo::common::ApolloApp {
 
   void SendCmd(ControlCommand *control_command);
 
- private:
+  private:
```
Comment (xiaoxq) [inline modules/control/control.h]: Should not change. Please run "apollo.sh lint" and fix all style issues.

File: modules/control/controller/BUILD [hunk 4]
```diff
@@ -189,5 +189,40 @@ cc_test(
         "@gtest//:main",
     ],
 )
-
+cc_binary(
```
Comment (xiaoxq) [inline modules/control/controller/BUILD]: Blank line before the section and new line after.

File: modules/control/controller/BUILD [hunk 5]
```diff
@@ -189,5 +189,40 @@ cc_test(
...
+        ":lat_controller",
+        "//modules/common:log",
+        "//modules/common/time",
+        "//modules/common/util",
+        "//modules/common/vehicle_state:vehicle_state_provider",
+        "//modules/control/proto:control_proto",
+        "//modules/planning/proto:planning_proto",
+        "@libprotobuf_mutator//:mutator",
+    ],
+    copts = ["-fsanitize=fuzzer,address",
```
Comment (xiaoxq) [inline modules/control/controller/BUILD]: List in rows like the "deps" section.

File: modules/control/controller/BUILD [hunk 6]
```diff
@@ -189,5 +189,40 @@ cc_test(
...
+        ":lon_controller",
+        "//modules/common:log",
+        "//modules/common/time",
+        "//modules/common/util",
+        "//modules/common/vehicle_state:vehicle_state_provider",
+        "//modules/control/proto:control_proto",
+        "//modules/planning/proto:planning_proto",
+        "@libprotobuf_mutator//:mutator",   
+    ],
+    copts = ["-fsanitize=fuzzer,address",
```
Comment (xiaoxq) [inline modules/control/controller/BUILD]: ditto

File: modules/control/controller/lat_controller_fuzzer.cc [hunk 7]
```diff
@@ -0,0 +1,147 @@
+/******************************************************************************
+ * Copyright 2017 The Apollo Authors. All Rights Reserved.
+ * Copyright 2018 Baidu X-Lab. Yunhan Jia <jiayunhan@baidu.com>
```
Comment (xiaoxq) [inline modules/control/controller/lat_controller_fuzzer.cc]: We would prefer unified copyright in the master repo.

File: WORKSPACE.in (hunk 2 above)
Comment (jiayunhan) [reply WORKSPACE.in]: I forked this repo from Google Github with some modifications, where do you suggest to put the repo? Is it possible to put it together with Apollo, under ApolloAuto. I will fix other issues. Thanks.

Comment (jiayunhan) [reply WORKSPACE.in]: Never mind, I will put it under our  Baidu X-Lab Github organization. Thanks.

**Comment by jiayunhan** [comment]
Made the required changes. Passed cpp lint, fixed copyright statements. Excludes all fuzz target from apollo.sh build, since they rely on clang. Currently they can only be built manually using bazel command. Moved the third-party repository to Baidu X-Lab organization.

**Pull request closed without merging**